compiler/grainformat/reformat.re

```diff
@@ -0,0 +1,2753 @@
+            ~expr=first,
+            ~parentIsArrow=false,
+            ~endChar=None,
+            ~original_source,
+            ~parent_loc,
+          ),
+          Doc.rparen,
+        ])
+
+      | _ =>
+        Doc.concat([
+          Doc.text(function_name),
+          print_expression(
+            ~expr=first,
+            ~parentIsArrow=false,
+            ~endChar=None,
+            ~original_source,
+            ~parent_loc,
+          ),
+        ])
+      };
+    } else {
+      Doc.text(
+        function_name
+        ++ "/* formatter error prefix should have 1 arg exactly */",
+      );
+    };
+  } else if (infixop(function_name)) {
+    let first = List.hd(expressions);
+    let second = List.hd(List.tl(expressions)); // assumes an infix only has two expressions
+    let first_brackets =
+      switch (first.pexp_desc) {
+      | PExpIf(_) =>
+        Doc.concat([
+          Doc.lparen,
+          print_expression(
+            ~expr=first,
+            ~parentIsArrow=false,
+            ~endChar=None,
+            ~original_source,
+            ~parent_loc,
+          ),
+          Doc.rparen,
+        ])
+      | PExpApp(fn, _) =>
+        let leftfn = get_function_name(fn);
+        if (infixop(leftfn)) {
+          Doc.concat([
+            Doc.lparen,
+            print_expression(
+              ~expr=first,
+              ~parentIsArrow=false,
+              ~endChar=None,
+              ~original_source,
+              ~parent_loc,
+            ),
+            Doc.rparen,
+          ]);
+        } else {
+          Doc.concat([
+            Doc.lparen,
+            print_expression(
+              ~expr=first,
+              ~parentIsArrow=false,
+              ~endChar=None,
+              ~original_source,
+              ~parent_loc,
+            ),
+            Doc.rparen,
+          ]);
+        };
+      | _ =>
+        print_expression(
+          ~expr=first,
+          ~parentIsArrow=false,
+          ~endChar=None,
+          ~original_source,
+          ~parent_loc,
+        )
+      };
+
+    let second_brackets =
+      switch (second.pexp_desc) {
+      | PExpIf(_)
+      | PExpApp(_) =>
+        Doc.concat([
+          Doc.lparen,
+          print_expression(
+            ~expr=second,
+            ~parentIsArrow=false,
+            ~endChar=None,
+            ~original_source,
+            ~parent_loc,
+          ),
+          Doc.rparen,
+        ])
+      | _ =>
+        print_expression(
+          ~expr=second,
+          ~parentIsArrow=false,
+          ~endChar=None,
+          ~original_source,
+          ~parent_loc,
+        )
+      };
+    Doc.concat([
+      first_brackets,
+      Doc.space,
+      Doc.text(function_name),
+      Doc.space,
+      second_brackets,
+    ]);
+  } else {
+    let func_name = get_function_name(func);
+
+    if (func_name == "[...]") {
+      resugar_list(~expressions, ~parent_loc, ~original_source);
+    } else if (func_name == "throw"
```

Comment:
Let's add a constant for these that we can check against.

Reply:
Good catch, done